### PR TITLE
Add a match priority check that compares items when no text is typed

### PR DESCRIPTION
**Pull request by rchande**

### CODE REVIEW

**Comment by rchande** [comment]
retest this please

**Comment by rchande** [comment]
@Pilchie Tests added

**Comment by rchande** [comment]
Addresses https://github.com/dotnet/roslyn/issues/15090

File: src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb [hunk 1]
```diff
@@ -1975,6 +1975,29 @@ class Program
         End Function
 
         <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function TargetTypePreselection_DoesNotOverrideEnumPreselection2() As Task
+            Using state = TestState.CreateCSharpTestState(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb]: There were a bunch of bugs filed.  Do we need additional tests for those cases?  Or are they all *for certain* the same issue? :)

**Comment by CyrusNajmabadi** [comment]
Even if it's a PITA, can you look at all the duped bugs, just to validate?

**Comment by rchande** [comment]
@CyrusNajmabadi I went through the 3 bugs I could find (https://github.com/dotnet/roslyn/issues/14813, https://github.com/dotnet/roslyn/issues/15090, and https://github.com/dotnet/roslyn/issues/15186) and they're all fixed by this change. Furthermore, those are all instances where enum completion is supposed to come up with nothing else typed, which is what I've added a test for. I think we're good to go.

**Comment by rchande** [comment]
Ping @dotnet/roslyn-ide. Any other thoughts? I'd like to take this to ask mode today.

**Comment by CyrusNajmabadi** [comment]
👍

**Comment by rchande** [comment]
**Ask Mode Info**

- **Customer scenario**: Enum preselection does not work if the completion list comes up before the user has typed any text
- **Bugs this fixes**: #14813, #15090, and #15186
- **Workarounds**: None
- **Risk**: Very low--deleted code is restored
- **Performance impact**: None.
- **Is this a regression from a previous update?*?: Yes, this was regressed in RC. https://github.com/dotnet/roslyn/pull/14668 removed an if statement that enabled completion to preselect items with nothing typed
- **How was the bug found?**: Dogfooding; customer feedback

**Comment by Pilchie** [comment]
Tagging @MattGertz  - we were hoping to get this for RC.2...

**Comment by MattGertz** [comment]
Yes, let's take this for Escrow on Tuesday.  This has had a goodly amount of customer feedback associated with it and so would be a good thing to check, even though it doesn't fit the usual Escrow bar.

File: src/Features/Core/Portable/Completion/CompletionHelper.cs [hunk 2]
```diff
@@ -183,6 +183,12 @@ public int CompareItems(CompletionItem item1, CompletionItem item2, string filte
                 return 1;
             }
 
+            var preselectionDiff = ComparePreselection(item1, item2);
```
Comment (rchande) [inline src/Features/Core/Portable/Completion/CompletionHelper.cs]: No, compareitems is used to choose the best one from the list.

**Pull request merged**